alf/trainers/evaluator.py

```diff
@@ -0,0 +1,253 @@
+    flags.DEFINE_string('conf', None, 'Path to the alf config file.')
+    flags.DEFINE_multi_string('conf_param', None, 'Config binding parameters.')
+
+
+FLAGS = flags.FLAGS
+
+
+def _worker(job_queue: mp.Queue,
+            done_queue: mp.Queue,
+            conf_file: str,
+            num_parallel_envs: int,
+            root_dir: str,
+            seed: Optional[int] = None):
+    try:
+        _define_flags()
+        FLAGS(sys.argv, known_only=True)
+        FLAGS.mark_as_parsed()
+        FLAGS.alsologtostderr = True
+        logging.set_verbosity(logging.INFO)
+        logging.get_absl_handler().use_absl_log_file(log_dir=root_dir)
+        logging.use_absl_handler()
+        if torch.cuda.is_available():
+            alf.set_default_device("cuda")
+        if seed is not None:
+            # seed the environments differently from the training
+            seed = seed + 13579
+        common.set_random_seed(seed)
+        alf.config('TrainerConfig', mutable=False, random_seed=seed)
+        if num_parallel_envs > 1:
+            alf.config(
+                'create_environment',
+                num_parallel_environments=num_parallel_envs,
+                mutable=False)
+        else:
+            alf.config('create_environment', nonparallel=True)
+        try:
+            common.parse_conf_file(conf_file)
+        except Exception as e:
+            alf.close_env()
+            raise e
+
+        config = policy_trainer.TrainerConfig(root_dir=root_dir)
+
+        env = alf.get_env()
+        env.reset()
+        data_transformer = create_data_transformer(
+            config.data_transformer_ctor, env.observation_spec())
+        config.data_transformer = data_transformer
+        # keep compatibility with previous gin based config
+        common.set_global_env(env)
+        observation_spec = data_transformer.transformed_observation_spec
+        common.set_transformed_observation_spec(observation_spec)
+
+        algorithm_ctor = config.algorithm_ctor
+        algorithm = algorithm_ctor(
+            observation_spec=observation_spec,
+            action_spec=env.action_spec(),
+            reward_spec=env.reward_spec(),
+            config=config)
+        algorithm.set_path('')
+        policy_trainer.Trainer._trainer_progress.set_termination_criterion(
+            config.num_iterations, config.num_env_steps)
+
+        eval_dir = os.path.join(root_dir, 'eval')
+        summary_writer = alf.summary.create_summary_writer(
+            eval_dir, flush_secs=config.summaries_flush_secs)
+        alf.summary.enable_summary()
+        logging.info("Evaluator started")
+        with alf.summary.push_summary_writer(summary_writer):
+            while True:
+                job = job_queue.get()
+                if job.type == "eval":
+                    logging.info("Start evaluation")
+                    alf.summary.set_global_counter(job.global_counter)
+                    policy_trainer.Trainer._trainer_progress.update(
```

Comment:
Why do we need to update trainer progress here?

Reply:
comment added.